BinaryObjectScanner/FileType/ISO9660.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using BinaryObjectScanner.Data;
+using SabreTools.Data.Models.ISO9660;
+using SabreTools.IO.Extensions;
+
+namespace BinaryObjectScanner.FileType
+{
+    /// <summary>
+    /// ISO9660
+    /// </summary>
+    public class ISO9660 : DiskImage<SabreTools.Serialization.Wrappers.ISO9660>
+    {
+        /// <inheritdoc/>
+        public ISO9660(SabreTools.Serialization.Wrappers.ISO9660 wrapper) : base(wrapper) { }
+
+        /// <inheritdoc/>
+        public override string? Detect(Stream stream, string file, bool includeDebug)
+        {
+            // Create the output dictionary
+            var protections = new ProtectionDictionary();
+
+            // Standard checks
+            var subProtections
+                = RunISOChecks(file, StaticChecks.ISO9660CheckClasses, includeDebug);
+            protections.Append(file, subProtections.Values);
+
+            // If there are no protections
+            if (protections.Count == 0)
+                return null;
+
+            // Create the internal list
+            var protectionList = new List<string>();
+            foreach (string key in protections.Keys)
+            {
+                protectionList.AddRange(protections[key]);
+            }
+
+            return string.Join(";", [.. protectionList]);
+        }
+
+        // Checks whether the sequence of bytes is pure data (as in, not empty, not text, just high-entropy data)
+        public static bool IsPureData(byte[] bytes)
+        {
+            // Check if there are three 0x00s in a row. Two seems like pushing it
+            byte[] containedZeroes = {0x00, 0x00, 0x00};
+            int index = 0;
+            for (int i = 0; i < bytes.Length; ++i) {
+                if (bytes[i] == containedZeroes[index])
+                {
+                    if (++index >= containedZeroes.Length)
+                        return false; 
+                }
+                else
+                    index = 0;
+            }
+            
+            // Checks if there are strings in the data
+            // TODO: is this too dangerous, or too faulty?
+            // Currently-found worst cases:
+            // "Y:1BY:1BC" in Redump ID 23339
+            var strings = bytes.ReadStringsWithEncoding(charLimit: 7, Encoding.ASCII);
+            Regex rgx = new Regex("[^a-zA-Z0-9 -'!?,.]");
```

Comment:
I'm interested in this logic as I'm not sure I understand how this determines pseudo-randomness

Reply:
I'm sure there's a better way to determine randomness/entropy, but I do not know what it is. If you have any suggestions, I would be more than happy to throw out everything in this function.

Reply:
Would `spamsum` be of any use here? That's inherently more a measure of randomness and "closeness" between data rather than a "true" checksum. I know you added the method to compare values but I don't know if there's something similar for analyzing how much entropy there is.

Reply:
That's incredibly unfortunate. If this works for now, then it can be kept. Investigating calculated entropy might be interesting for the future, though.